test/integration/apiserver/apply/apply_test.go

```diff
@@ -3408,3 +3408,206 @@ func TestSubresourceField(t *testing.T) {
+					"finalizers": [
+						"kubernetes.io/pv-protection"
+					],
+					"labels": {
+						"type": "local"
+					},
+					"managedFields": [
+						{
+							"apiVersion": "v1",
+							"fieldsType": "FieldsV1",
+							"fieldsV1": {
+								"f:metadata": {
+									"f:labels": {
+										"f:type": {}
+									}
+								},
+								"f:spec": {
+									"f:accessModes": {},
+									"f:capacity": {
+										"f:storage": {}
+									},
+									"f:hostPath": {
+										"f:path": {}
+									},
+									"f:storageClassName": {}
+								}
+							},
+							"manager": "apply_test",
+							"operation": "Apply",
+							"time": "2022-06-08T23:46:32Z"
+						},
+						{
+							"apiVersion": "v1",
+							"fieldsType": "FieldsV1",
+							"fieldsV1": {
+								"f:status": {
+									"f:phase": {}
+								}
+							},
+							"manager": "kube-controller-manager",
+							"operation": "Update",
+							"subresource": "status",
+							"time": "2022-06-08T23:46:32Z"
+						},
+						{
+							"apiVersion": "v1",
+							"fieldsType": "FieldsV1",
+							"fieldsV1": {
+								"f:spec": {
+									"f:claimRef": {}
+								}
+							},
+							"manager": "kube-controller-manager",
+							"operation": "Update",
+							"time": "2022-06-08T23:46:37Z"
+						}
+					],
+					"name": "pv-storage",
+					"uid": "112b18f7-fde6-4e48-aa61-f5168bd576b8"
+				},
+				"spec": {
+					"accessModes": [
+						"ReadWriteOnce"
+					],
+					"capacity": {
+						"storage": "16Mi"
+					},
+					"claimRef": {
+						"apiVersion": "v1",
+						"kind": "PersistentVolumeClaim",
+						"name": "pvc-storage",
+						"namespace": "default",
+						"resourceVersion": "15499",
+						"uid": "2018e302-7b12-406c-9fa2-e52535d29e48"
+					},
+					"hostPath": {
+						"path": "“/tmp/mydata",
+						"type": ""
+					},
+					"persistentVolumeReclaimPolicy": "Retain",
+					"volumeMode": "Filesystem"
+				},
+				"status": {
+					"phase": "Bound"
+				}
+			}
+`)
+
+	// Re-applies name and namespace
+	originalPV := []byte(`
+	{
+		"kind": "PersistentVolume",
+		"apiVersion": "v1",
+		"metadata": {
+			"labels": {
+				"type": "local"
+			},
+			"name": "pv-storage",
+		},
+		"spec": {
+			"storageClassName": "",
+			"capacity": {
+				"storage": "16Mi"
+			},
+			"accessModes": [
+				"ReadWriteOnce"
+			],
+			"hostPath": {
+				"path": "“/tmp/mydata"
+			},
+			"claimRef": {
+				"name": "pvc-storage",
+				"namespace": "default"
+			}
+		},
+	}
+	`)
+
+	// oldPersistentVolume, err := yamlutil.ToJSON(oldPersistentVolume)
+	// if err != nil {
+	// 	t.Fatalf("Failed marshal original yaml: %v", err)
+	// }
+
+	// originalPV, err = yamlutil.ToJSON(originalPV)
+	// if err != nil {
+	// 	t.Fatalf("Failed marshal yaml: %v", err)
+	// }
+
+	// Write our raw object in with its managed fields
+	originalObj, err := client.CoreV1().RESTClient().
+		Post().
+		Param("fieldManager", "apply_test").
+		Resource("persistentvolumes").
+		Body(oldPersistentVolume).
```

Comment:
I'm surprised this actually worked, trying to create an object with an existing uid set. Did you verify the persisted/returned object had the managedFields set you expected?

Reply:
Yeah, I would expect all the fields set in the request to be owned by the POSTer here.

Reply:
I think to be sure that you have the managedfields that you want, it'd be simpler to:
1. Create the object,
2. Send a patch (non-apply) that changes nothing in the object but the managedFields

Reply:
Added separate update for managed fields. also added checks at the end to verify the managed fields were to be expected